Classes/BAPromise.h

```diff
@@ -9,82 +9,82 @@
 #import <Foundation/Foundation.h>
 
 /* block definitions */
-typedef void (^BAPromiseOnFulfillBlock)(id value);
-typedef void (^BAPromiseOnRejectedBlock)(NSError *error);
-typedef id (^BAPromiseThenRejectedBlock)(NSError *error);
+typedef void (^BAPromiseOnFulfillBlock)(id _Nullable value);
+typedef void (^BAPromiseOnRejectedBlock)(NSError * _Nullable error);
```

Comment:
rejected blocks should not be called with anything other than a valid NSError (however, some more changes will have to be done to support this)